jabgui/src/main/java/org/jabref/gui/libraryproperties/general/GeneralPropertiesViewModel.java

```diff
@@ -196,24 +196,40 @@ public StringProperty laTexFileDirectoryProperty() {
     }
 
     private Path getBrowseDirectory(String configuredDir) {
+        Optional<Path> libPath = this.databaseContext.getDatabasePath();
+        Path workingDir = preferences.getFilePreferences().getWorkingDirectory();
+
+        if (libPath.isEmpty()) {
+            Path potetialAbsolutePath = Path.of(configuredDir);
+            return Files.isDirectory(potetialAbsolutePath) ? potetialAbsolutePath : workingDir;
+        }
         if (configuredDir.isEmpty()) {
-            return preferences.getFilePreferences().getWorkingDirectory();
+            return workingDir;
         }
-        Optional<Path> foundPath = this.databaseContext.getFileDirectories(preferences.getFilePreferences()).stream()
-                                                       .filter(path -> path.toString().endsWith(configuredDir))
-                                                       .filter(Files::exists).findFirst();
 
-        if (foundPath.isEmpty()) {
-            dialogService.notify(Localization.lang("Path %0 could not be resolved. Using working dir.", configuredDir));
-            return preferences.getFilePreferences().getWorkingDirectory();
+        Path configuredPath = libPath.get().getParent().resolve(configuredDir).normalize();
+
+        // As the user types, the configureDir may become invalid.
```

Comment:
The comment is a bit misleading - I think you mean "configDir can be input manually and thus be invalid"